src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Repositories/DelegationMetadataRepository.cs

```diff
@@ -40,31 +40,34 @@ public class DelegationMetadataRepository : IDelegationMetadataRepository
                 postgresSettings.Value.AuthorizationDbPwd);
+            NpgsqlConnection.GlobalTypeMapper.MapEnum<DelegationChangeType>("delegation.delegationchangetype");
         }
 
         /// <inheritdoc/>
-        public async Task<bool> InsertDelegation(string altinnAppId, int offeredByPartyId, int? coveredByPartyId, int? coveredByUserId, int delegatedByUserId, string blobStoragePolicyPath, string blobStorageVersionId, bool isDeleted = false)
+        public async Task<DelegationChange> InsertDelegation(DelegationChange delegationChange)
         {
             try
             {
                 using NpgsqlConnection conn = new NpgsqlConnection(_connectionString);
                 await conn.OpenAsync();
 
-                NpgsqlCommand pgcom = new NpgsqlCommand(insertDelegationChangeSql, conn);
-                pgcom.Parameters.AddWithValue("_altinnAppId", altinnAppId);
-                pgcom.Parameters.AddWithValue("_offeredByPartyId", offeredByPartyId);
-                pgcom.Parameters.AddWithValue("_coveredByUserId", coveredByUserId.HasValue ? coveredByUserId.Value : DBNull.Value);
-                pgcom.Parameters.AddWithValue("_coveredByPartyId", coveredByPartyId.HasValue ? coveredByPartyId.Value : DBNull.Value);
-                pgcom.Parameters.AddWithValue("_performedByUserId", delegatedByUserId);
-                pgcom.Parameters.AddWithValue("_blobStoragePolicyPath", blobStoragePolicyPath);
-                pgcom.Parameters.AddWithValue("_blobStorageVersionId", blobStorageVersionId);
-                pgcom.Parameters.AddWithValue("_isDeleted", isDeleted);
-
-                pgcom.Parameters.AddWithValue("_delegationChangeId", 0); // Must be included since it's specified in stored proc, but so far unable to get inserted value back.
-
-                await pgcom.ExecuteNonQueryAsync();
+                NpgsqlCommand pgcom = new NpgsqlCommand(insertDelegationChangeFunc, conn);
+                pgcom.Parameters.AddWithValue("_delegationChangeType", delegationChange.DelegationChangeType);
+                pgcom.Parameters.AddWithValue("_altinnAppId", delegationChange.AltinnAppId);
+                pgcom.Parameters.AddWithValue("_offeredByPartyId", delegationChange.OfferedByPartyId);
+                pgcom.Parameters.AddWithValue("_coveredByUserId", delegationChange.CoveredByUserId.HasValue ? delegationChange.CoveredByUserId.Value : DBNull.Value);
+                pgcom.Parameters.AddWithValue("_coveredByPartyId", delegationChange.CoveredByPartyId.HasValue ? delegationChange.CoveredByPartyId.Value : DBNull.Value);
+                pgcom.Parameters.AddWithValue("_performedByUserId", delegationChange.PerformedByUserId);
+                pgcom.Parameters.AddWithValue("_blobStoragePolicyPath", delegationChange.BlobStoragePolicyPath);
+                pgcom.Parameters.AddWithValue("_blobStorageVersionId", delegationChange.BlobStorageVersionId);
+
+                using NpgsqlDataReader reader = await pgcom.ExecuteReaderAsync();
```

Comment:
```suggestion
                await using NpgsqlDataReader reader = await pgcom.ExecuteReaderAsync();
```

`NpgsqlCommand` implementerer  `IAsyncDisposable`, men ser at at `DisposeAsync` bare kaller `Dispose`, så det har neppe noen stor praktisk betydning.

Reply:
haven't seen that recommendation myself but fixed 👍